lib/timetable-form/useValidation.tsx

```diff
@@ -43,5 +52,56 @@ export const useValidation = (): {
     []
   );
 
-  return { errors, validateDevelopmentPlanEventDates };
+  const validateDevelopmentPlanDescription = useCallback(
+    (developmentPlan: Pick<DevelopmentPlan, "description">) => {
+      const errors: ValidationErrorItem[] = [];
+
+      const validationResult = developmentPlanSchema.validate(developmentPlan, {
+        abortEarly: false,
+      });
+
+      if (validationResult.error) {
+        const validationErrors = validationResult.error.details.map(
+          (error) => ({
+            ...error,
+            path: [validationResult.value.reference, ...error.path],
+          })
+        );
+        errors.push(...validationErrors);
+      }
+
+      setErrors(errors);
+    },
+    []
+  );
+
+  const validateDevelopmentPlanName = useCallback(
+    (developmentPlan: Pick<DevelopmentPlan, "name">) => {
+      const errors: ValidationErrorItem[] = [];
+
+      const validationResult = developmentPlanSchema.validate(developmentPlan, {
+        abortEarly: false,
+      });
+
+      if (validationResult.error) {
+        const validationErrors = validationResult.error.details.map(
+          (error) => ({
+            ...error,
+            path: [validationResult.value.reference, ...error.path],
+          })
+        );
+        errors.push(...validationErrors);
+      }
+
+      setErrors(errors);
+    },
+    []
+  );
+
+  return {
+    errors,
+    validateDevelopmentPlanEventDates,
+    validateDevelopmentPlanName,
+    validateDevelopmentPlanDescription,
```

Comment:
We might be able to share some of the code but I think validation for different pages belongs in different files. We could restructure a little, maybe something like this?

```
|-- pages
|---- stage-page
|------ StagePage.tsx
|------ StagePage.test.tsx
|------ useStagePageValidation.ts
|---- description-page
|------ DescriptionPage.tsx
|------ DescriptionPage.test.tsx
|------ useDescriptionPageValidation.ts
...
```

Reply:
We'd probably also want to split apart files with the validation rules

Reply:
Leave this for now as I think we'll have to give this some more thought with the routing